hal/sleep_api.h

```diff
@@ -29,13 +29,15 @@ extern "C" {
 
 /** Send the microcontroller to sleep
  *
- * The processor is setup ready for sleep, and sent to sleep using __WFI(). In this mode, the
+ * The processor is setup ready for sleep, and sent to sleep. In this mode, the
  * system clock to the core is stopped until a reset or an interrupt occurs. This eliminates
  * dynamic power used by the processor, memory systems and buses. The processor, peripheral and
  * memory state are maintained, and the peripherals continue to work and can generate interrupts.
  *
  * The processor can be woken up by any internal peripheral interrupt or external pin interrupt.
  *
+ * The wake-up time should be less than 10 us.
```

Comment:
Well we are writing a specification, if we give a recommendation we can't rely on the implementation's behavior.

> Could we isolate and make it point on it's own requirements wake up sources.
I don't follow this one, please elaborate.

Ignore that, it is clearly mentioned after the main description body, I missed it somehow.